dbt/adapters/bigquery/connections.py

```diff
@@ -292,8 +298,29 @@ def exception_handler(self, sql):
                 exc_message = exc_message.split(BQ_QUERY_JOB_SPLIT)[0].strip()
             raise DbtRuntimeError(exc_message)
 
-    def cancel_open(self) -> None:
-        pass
+    def cancel_open(self) -> List[str]:
+        names = []
+        this_connection = self.get_if_exists()
+        with self.lock:
+            for thread_id, connection in self.thread_connections.items():
+                if connection is this_connection:
+                    continue
+
+                if connection.handle is not None and connection.state == ConnectionState.OPEN:
```

Comment:
This could be over-engineering, but I would consider putting the contents of this if block into its own method.

Reply:
to be clear are you referring to lines 308-321?

Reply:
I'm referring to lines 310-318. Then this would look like:

```python
names = []
this_connection = self.get_if_exists()

with self.lock:
    for thread_id, connection in self.thread_connections.items():

        if connection is this_connection:
            continue

        if connection.handle and connection.state == ConnectionState.OPEN:
            self.close_thread(thread_id, connection)  # or whatever name you choose

        if name := connection.name:
            names.append(name)

return names
```

Something else worth considering is whether we want to handle all of the threads within a connection. I don't know if there's more than one thread for a connection, but I feel like there is. If there's a connection with more than one thread, you'll close that connection in the second condition above when you get to the first thread. Then you'll skip past the second condition for every other thread since `connection.state` should be closed at that point.

I think what you probably want is a list of job_ids by connection. Then for each connection you would cancel the job. Once all jobs are cancelled, then close the connection.